src/Evolution/Systems/Cce/WorldtubeBufferUpdater.cpp

```diff
@@ -420,13 +480,281 @@ double MetricWorldtubeH5BufferUpdater<T>::update_buffers_for_time(
+        "Time must not vary fastest when using ADM quantities for the metric "
+        "worldtube buffer updater.");
+  }
+  const size_t number_of_times = time_span_end - time_span_start;
+  const size_t size_of_individual_time =
+      Spectral::Swsh::number_of_swsh_collocation_points(l_max_);
+  const size_t size_of_full_buffer =
+      get<0, 0>(get<Tags::detail::SpatialMetric<DataVector>>(*buffers)).size();
+
+  TempBuffer<tmpl::list<
+      ::Tags::Tempi<0, 3>,
+      ::Tags::Tempi<1, 2, ::Frame::Spherical<::Frame::Inertial>>,
+      ::Tags::Tempi<2, 3>, ::Tags::TempiJ<0, 3>, ::Tags::Tempijj<0, 3>,
+      ::Tags::Tempii<0, 3>, ::Tags::TempScalar<0>, ::Tags::TempScalar<1>,
+      ::Tags::TempII<0, 3>, ::Tags::TempI<0, 3>, ::Tags::TempI<1, 3>>>
+      temp_buffer{size_of_full_buffer};
+  auto& cartesian_cauchy_coordinates_buffer =
+      get<::Tags::Tempi<0, 3>>(temp_buffer);
+  auto& unused_angular_cauchy_coordinates_buffer =
+      get<::Tags::Tempi<1, 2, ::Frame::Spherical<::Frame::Inertial>>>(
+          temp_buffer);
+
+  // Have to compute angular coordinates for each time independently
+  for (size_t i = 0; i < number_of_times; i++) {
+    tnsr::i<DataVector, 3> cartesian_cauchy_coordinates{};
+    tnsr::i<DataVector, 2, ::Frame::Spherical<::Frame::Inertial>>
+        unused_angular_cauchy_coordinates{};
+    // NOLINTBEGIN
+    for (size_t j = 0; j < 3; j++) {
+      cartesian_cauchy_coordinates[j].set_data_ref(
+          cartesian_cauchy_coordinates_buffer[j].data() +
+              i * size_of_individual_time,
+          size_of_individual_time);
+      if (j < 2) {
+        unused_angular_cauchy_coordinates[j].set_data_ref(
+            unused_angular_cauchy_coordinates_buffer[j].data() +
+                i * size_of_individual_time,
+            size_of_individual_time);
+      }
+      // NOLINTEND
+    }
+
+    // Calling this function is easier than calling the individual parts
+    // that only compute the cartesian coords
+    Spectral::Swsh::create_angular_and_cartesian_coordinates(
+        make_not_null(&cartesian_cauchy_coordinates),
+        make_not_null(&unused_angular_cauchy_coordinates), l_max_);
+  }
+
+  // Need lots of temporaries to convert quantities
+  auto& cartesian_deriv_lapse = get<::Tags::Tempi<2, 3>>(temp_buffer);
+  auto& cartesian_deriv_shift = get<::Tags::TempiJ<0, 3>>(temp_buffer);
+  auto& cartesian_deriv_spatial_metric =
+      get<::Tags::Tempijj<0, 3>>(temp_buffer);
+  auto& extrinsic_curvature = get<::Tags::Tempii<0, 3>>(temp_buffer);
+  auto& trace_extrinsic_curvature = get<::Tags::TempScalar<0>>(temp_buffer);
+  auto& determinant_spatial_metric = get<::Tags::TempScalar<1>>(temp_buffer);
+  auto& inverse_spatial_metric = get<::Tags::TempII<0, 3>>(temp_buffer);
+  auto& auxiliary_shift = get<::Tags::TempI<0, 3>>(temp_buffer);
+  auto& conformal_christoffel = get<::Tags::TempI<1, 3>>(temp_buffer);
+
+  const auto set_radial_deriv =
+      [&](const gsl::not_null<DataVector*> radial_deriv,
```

Comment:
`&cartesian_cauchy_coordinates_buffer`. Capturing everything is generally strongly recommended against since it can trivially lead to dangling references and makes the closure generated a lot more complicated